### ISSUE TITLE
Hosting profiles on internal web server

### ISSUE THREAD

**Issue by r0b1nv4np** [google/rekall-profiles#19]
Hi, 

I want to be able to host the profiles repo on an internal web server due to github being blocked by company policy. 

I have cloned the repo and started a simple web server locally to test to check if rekall can use the profiles from the server I have hosted. 

However, starting the rekall leads to an error:

`rekall -v live --repository http://127.0.01`
`WARNING:rekall.1:Inventory for repository "http://127.0.0.1" seems malformed. Are you behind a captive portal or proxy? If this is a custom repository, did you forget to create an inventory? You must use the tools/profiles/build_profile_repo.py tool with the --inventory flag `

1. Is it possible for me to host the repo locally and serve it via a web server?
2. I could not find the file build_profile_repo.py in the tools folder in rekall repo either. 

If I am making an elementary mistake, I apologise in advance. 

Any help would be much appreciated.

**Comment by scudette** [comment]
Yes you can serve the profiles locally. You just need to check them out from GIT then serve them. Here is an example:

```
~/rekall-profiles⟫ python -m SimpleHTTPServer 8000
Serving HTTP on 0.0.0.0 port 8000 ...
127.0.0.1 - - [26/Mar/2017 22:48:55] "GET //v1.0/inventory.gz HTTP/1.1" 200 -
127.0.0.1 - - [26/Mar/2017 22:48:55] "GET //v1.0/Linux/index.gz HTTP/1.1" 200 -
127.0.0.1 - - [26/Mar/2017 22:48:55] "GET //v1.0/nt/eprocess_index.gz HTTP/1.1" 200 -
127.0.0.1 - - [26/Mar/2017 22:48:55] "GET //v1.0/nt/index.gz HTTP/1.1" 200 -
127.0.0.1 - - [26/Mar/2017 22:48:56] "GET //v1.0/OSX/index.gz HTTP/1.1" 200 -
127.0.0.1 - - [26/Mar/2017 22:48:56] "GET //v1.0/pe.gz HTTP/1.1" 200 -
127.0.0.1 - - [26/Mar/2017 22:48:56] "GET //v1.0/nt/GUID/D7EA2B6682984A0E8697620F5571B7BF2.gz HTTP/1.1" 200 -
127.0.0.1 - - [26/Mar/2017 22:48:56] "GET //v1.0/nt/undocumented.gz HTTP/1.1" 200 -
```

And you launch Rekall with the --repository_path flag:
```
⟫ rekal -v -f ~/images/myimage.aff4 --repository_path http://127.0.0.1:8000/  pslist
```

The build_profile_repository was an early script which is now replaced by the `manage_repo` plugin.

**Comment by r0b1nv4np** [comment]
This worked ! I was infact making an elementary mistake. 

Thank you for your time.

**Comment by r0b1nv4np** [comment]
New issue here:

I added a self-signed cert to my webserver hosting the profiles repo and set the repository_path to https://<websever>. 

I have double checked all other related settings and everything looks ok.

Is there a certificate issue? 

How does rekall access github over SSL  but refuses to do the same on against an internal web server over ssl with a self-signed cert? 

Do let me know.

**Comment by scudette** [comment]
Rekall uses the requests library to access HTTPS URL. This checks certs by
default. See here:

http://docs.python-requests.org/en/master/user/advanced/#ssl-cert-verification

If you want to set your own self signed cert you will need to add the
signer to the root CA bundle and pass it in the REQUESTS_CA_BUNDLE
environment variable.

Thanks
Michael.

**Comment by r0b1nv4np** [comment]
This worked for me ! 

To quickly highlight the steps in case someone comes across this:

I exported the cert and saved it. 
I launched mmc and add it to the root CA bundle using the process described here:
https://technet.microsoft.com/en-us/library/cc754841(v=ws.11).aspx  (Adding certificates to the Trusted Root Certification Authorities store for a local computer)

Thank you !
